### PR TITLE
Make the API able to take an array for name matching

### PR DESCRIPTION
**Pull request by metype**
Introduces no breaking changes and old queries behave exactly the same, but if you pass an array into `searchPlayerName` in a `players` query, it will now do a bit more matching work. 

Added a new parameter for `players` queries, `matchAll` to go along with allowing for arrays into `searchPlayerName`. If `matchAll` is false, any of the strings inside `searchPlayerName` can match a name for it to be returned, otherwise they must all match.

### CODE REVIEW

File: src/models/Player.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ const { Op } = Sequelize;
 
 class Player {
   static async getPlayerList({
-    limit, offset, order, searchPlayerName,
+    limit, offset, order, matchAll, searchPlayerName,
```
Comment (Trikolon) [inline src/models/Player.js:10]: Could you add some jsdoc for this method that explains what at the args are? `matchAll` in itself is not very descriptive.

File: src/models/Player.js [hunk 2]
```diff
@@ -16,12 +16,25 @@ class Player {
...
 
     if (searchPlayerName) {
-      options.where = {
-        lastSeenName: Sequelize.where(
-          Sequelize.fn('LOWER',
-            Sequelize.col('BAT_player')), 'LIKE', searchPlayerName,
-        ),
-      };
+
+        var nameMatchElements = searchPlayerName.map(nameElement => {
```
Comment (Trikolon) [inline src/models/Player.js:20]: Please use `const` instead of `var` to follow the code style of this project.

File: src/models/Player.js [hunk 3]
```diff
@@ -16,12 +16,25 @@ class Player {
...
+            return {
+                lastSeenName: { 
+                    [Op.like] : nameElement,
+                }
+            }
+        })
+        
+        // And forces all to match, Or allows any to match
+        if(matchAll) {
```
Comment (Trikolon) [inline src/models/Player.js:29]: can you run an `eslint --fix`? This looks like it could be auto formatted.

File: src/models/Player.js [hunk 4]
```diff
@@ -70,4 +83,4 @@ class Player {
   }
 }
 
-module.exports = { Player };
+module.exports = { Player };
```
Comment (Trikolon) [inline src/models/Player.js:86]: unrelated change?

**Review by Trikolon — CHANGES_REQUESTED** [review]
Thanks! I think this would benefit from a bit more documentation.

File: src/models/Player.js (hunk 4 above)
Comment (metype) [reply src/models/Player.js:86]: This is unrelated, really unsure why this happened or is changed.

File: src/models/Player.js [hunk 5]
```diff
@@ -6,6 +6,14 @@ const { isNullUUID, fixUpUUID, stripUUID } = require('./util');
 const { Op } = Sequelize;
 
 class Player {
+  /**
```
Comment (Trikolon) [inline src/models/Player.js:9]: nit: alignment?

File: src/models/Player.js [hunk 6]
```diff
@@ -6,6 +6,14 @@ const { isNullUUID, fixUpUUID, stripUUID } = require('./util');
...
+ * Returns a list of results from a players query in result and the number of total matches in totalCount.
+ * @param {number} limit - Max number of results to return.
+ * @param {number} offset - Number to offset the first index of results by, useful for pagination.
+ * @param {string} order - ASC or DESC, determines the order of players returned in result ordered by first joined date.
+ * @param {boolean} matchAll - If true; searchPlayerName's elements all should match lastSeenName to be returned. If false, any can match.
+ * @param {string[]} searchPlayerName - List of strings to use for filtering by lastSeenName
```
Comment (Trikolon) [inline src/models/Player.js:15]: nit: might be good to mention that these args are optional. you can do that in jsdoc like so:

```
@param {boolean} [matchAll=false] - foo bar

File: src/typeDefs/query.js [hunk 7]
```diff
@@ -21,8 +21,10 @@ const query = gql`
             offset: Int = 0,
             "Whether to sort ascending or descending by player first join date."
             order: ListOrder = DESC,
-            "Filter results by last seen player name. Supports SQL LIKE patterns."
-            searchPlayerName: String
+            "If true, searchPlayerName must have all elements match a username, if false any can match."
+            matchAll: Boolean = true,
+            "Filter results by last seen player name. Supports SQL LIKE patterns. Matching parameters defined by matchAll"
+            searchPlayerName: [String]
```
Comment (Trikolon) [inline src/typeDefs/query.js:27]: This breaks backwards compatibility, correct? I *think* this is fine because we only use this query for players.tallcraft.com. In the future we should consider doing versioned endpoints of the API.

**Review by Trikolon — APPROVED** [review]
Thanks!

File: src/typeDefs/query.js (hunk 7 above)
Comment (metype) [reply src/typeDefs/query.js:27]: While versioning would be a good idea, backwards compatibility actually is not broken by this change. Passing a single string into a string[] field just passes a one-element length array and so it behaves exactly the same with the same old queries. I tested it.
